chalice/test.py

```diff
@@ -0,0 +1,326 @@
+                'MessageId': 'abcdefgh-51e4-5ae2-9964-b296c8d65d1a',
+                'Signature': 'signature',
+                'SignatureVersion': '1',
+                'SigningCertUrl': (
+                    'https://sns.us-west-2.amazonaws.com/cert.pem'),
+                'Subject': subject,
+                'Timestamp': '2018-06-26T19:41:38.695Z',
+                'TopicArn': 'arn:aws:sns:us-west-2:12345:TopicName',
+                'Type': 'Notification',
+                'UnsubscribeUrl': 'https://unsubscribe-url/'
+            }
+        }]}
+        return sns_event
+
+    def generate_s3_event(self, bucket, key, event_name='ObjectCreated:Put'):
+        # type: (str, str, str) -> Dict[str, Any]
+        s3_event = {
+            'Records': [
+                {'awsRegion': 'us-west-2',
+                 'eventName': event_name,
+                 'eventSource': 'aws:s3',
+                 'eventTime': '2018-05-22T04:41:23.823Z',
+                 'eventVersion': '2.0',
+                 'requestParameters': {'sourceIPAddress': '1.1.1.1'},
+                 'responseElements': {
+                     'x-amz-id-2': 'request-id-2',
+                     'x-amz-request-id': 'request-id-1'
+                 },
+                 's3': {
+                     'bucket': {
+                         'arn': 'arn:aws:s3:::%s' % bucket,
+                         'name': bucket,
+                         'ownerIdentity': {
+                             'principalId': 'ABCD'
+                         }
+                     },
+                     'configurationId': 'config-id',
+                     'object': {
+                         'eTag': 'd41d8cd98f00b204e9800998ecf8427e',
+                         'key': key,
+                         'sequencer': '005B039F73C627CE8B',
+                         'size': 0
+                     },
+                     's3SchemaVersion': '1.0'
+                 },
+                 'userIdentity': {'principalId': 'AWS:XYZ'}
+                 }
+            ]
+        }
+        return s3_event
+
+    def generate_sqs_event(self, message_bodies, queue_name='queue-name'):
+        # type: (str, str) -> Dict[str, Any]
+        records = [{
+            'attributes': {
+                'ApproximateFirstReceiveTimestamp': '1530576251596',
+                'ApproximateReceiveCount': '1',
+                'SenderId': 'sender-id',
+                'SentTimestamp': '1530576251595'
+            },
+            'awsRegion': 'us-west-2',
+            'body': body,
+            'eventSource': 'aws:sqs',
+            'eventSourceARN': 'arn:aws:sqs:us-west-2:12345:%s' % queue_name,
+            'md5OfBody': '754ac2f7a12df38320e0c5eafd060145',
+            'messageAttributes': {},
+            'messageId': 'message-id',
+            'receiptHandle': 'receipt-handle'
+        } for body in message_bodies]
+        sqs_event = {'Records': records}
+        return sqs_event
+
+    def generate_cw_event(self, source, detail_type,
```

Comment:
For scheduled events, are you supposed to use this method to get a sample event? I wonder if it makes sense to have a separate method for it.

Reply:
Yeah the intent is to use this method.  It's the same thing as a cloudwatch event without any additional detail (https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#schedule_event_type):

```
{
  "id": "53dc4d37-cffa-4f76-80c9-8b7d4a4d2eaa",
  "detail-type": "Scheduled Event",
  "source": "aws.events",
  "account": "123456789012",
  "time": "2019-10-08T16:53:06Z",
  "region": "us-east-1",
  "resources": [ "arn:aws:events:us-east-1:123456789012:rule/MyScheduledRule" ],
  "detail": {}
}
```